03-task-manager/starter/QuizAnswers2.txt

```diff
@@ -0,0 +1,19 @@
+1. In this lesson, you created a middleware function called asyncWrapper. Why?
+
+==>> Because this middleware function will wrap try/catch block so that controller functions will not repeat try/catch block. Instead these controller functions will call asyncWrapper to execute operations, which are asynchonous operations.
+
+2. Suppose that you want to make sure that both a status code and an error message are sent back to the user when they request the URL for a task that does not exist. Assume that you’ve created a CustomAPIError class and an error handler that references that class. Complete the code:
+
+==>> 
+const getTask = asyncWrapper(async (req, res, next) => {
+    const { id: taskID } = req.params;
+
+    const task = await Task.findOne({ _id: taskID });
+
+    if (!task) {
+        // your code here  
+        return next(createCustomError(`No task with id: ${taskID}`, 404));
```

Comment:
Instead of `next`, I think you want to call `res.send` here.

Reply:
Hi Alex,
The assignment question 2:

"Suppose that you want to make sure that both a status code and an error message are sent back to the user when they request the URL for a task that does not exist"

So I think it asks for knowledge checking.

However, as your recommendation, I would use this command:
==>> return res.status(404).json({ "No task with id": taskID });

Because we should send back to the Client that the task does not exist.

Please let me know if this is fine to you or I need to do another way.

By the way, do I need to update my assignment with your recommendation ? Because I got the email said that "Assignment Successful".

Reply:
Hi Will, sorry for the delay in responding to your comment. No need to update your answer as this is an optional assignment. Good work!